v2/contribute/decisions/thirdparty/0006-pre-built-ui-selection.mdx

```diff
@@ -0,0 +1,99 @@
+---
+id: "0006"
+title: Recipe selection for pre-built UI
+hide_title: true
+---
+
+import DecisionHeader from "/src/components/decisionLogs/DecisionHeader"
+import ArgumentPro from "/src/components/decisionLogs/ArgumentPro"
+import ArgumentCon from "/src/components/decisionLogs/ArgumentCon"
+import ArgumentNeut from "/src/components/decisionLogs/ArgumentNeut"
+
+# Recipe selection for pre-built UI
+
+<DecisionHeader status="proposed" lastUpdate="2022-12-08" created="2022-12-08" deciders={["rishabhpoddar", "sattvikc"]} proposedBy={["sattvikc"]} />
+
+## Context and Problem Statement
+
+In case of `usesDynamicLoginMethods` set to true, the backend responds with the recipes that are enabled in the core for a particular tenant, such as:
+```json
+{
+  "thirdParty": {
+    "enabled": true/false,
+    ...
+  },
+  "emailpassword": {
+    "enabled": true/false,
+    ...
+  },
+  "passwordless": {
+    "enabled": true/false,
+    ...
+  }
+}
+```
+
+Along with this, the user would have statically initialized a combination of auth recipes, such as emailpassword & thirdpartypasswordless, etc.
+
+Based on the information provided by backend (core) and the recipies initialized by the user, the SDK needs to decide which recipes must be used for the UI
+
+## Considered Options
+
+* Automatically create a recipe combination based on the recipies enabled in the core if not statically declared
+* Logic described below
+
+## Automatically create a recipe combination based on the recipies enabled in the core if not statically declared
+
+In this approach, if the user has not statically initialized a matching recipe, the SDK will automatically create a recipe combination based on the recipies enabled in the core.
```

Comment:
```suggestion
In this approach, if the user has not statically initialized a matching recipe, the SDK will automatically create a recipe combination based on the recipies enabled in the core. This also implies that we would get rid of thirdpartyemailpassword and thirdpartypasswordless and instead build the UI in a way that would allow for any combination of emailpassword, thirdparty and passwordless recipes used together.

```